src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java

```diff
@@ -199,9 +209,49 @@ protected HttpExchange() {
      *                       specified and no response body may be written.
      * @throws IOException   if the response headers have already been sent or an I/O error occurs
      * @see   HttpExchange#getResponseBody()
+     * @see   HttpExchange#CHUNKED_CONTENT
+     * @see   HttpExchange#NO_CONTENT
      */
     public abstract void sendResponseHeaders(int rCode, long responseLength) throws IOException;
 
+    /**
+     * Convenience method to send the response headers with no response body.
+     * The exchange is also closed.
+     * @param code the response code to send
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final void sendResponseHeadersNoContent(int code) throws IOException {
+        sendResponseHeaders(code, NO_CONTENT);
+        close();
+    }
```

Comment:
I wonder if this method should better be called `sendResponseNoContent(int code)`; Maybe wait to see if you get further suggestions.